### PR TITLE
[node, client] Swap both ways

### PR DESCRIPTION
**Pull request by rhlsthrm**
DAI <> ETH swaps enabled on node, not implemented in client.

### CODE REVIEW

File: modules/types/src/index.ts [hunk 1]
```diff
@@ -501,8 +512,8 @@ export function convertSwapParameters<To extends NumericTypeName>(
   to: To,
   obj: SwapParameters<any>,
 ): SwapParameters<NumericTypes[To]> {
-  const fromType = getType(obj.swapRate);
-  return convertFields(fromType, to, ["swapRate", "amount"], obj);
+  const fromType = getType(obj.amount);
```
Comment (LayneHaber) [inline modules/types/src/index.ts]:
```typescript
return convertAmountField(to, obj);
```

File: modules/node/src/appRegistry/appRegistry.service.ts [hunk 2]
```diff
@@ -150,6 +133,9 @@ export class AppRegistryService implements OnModuleInit {
     data: ProposeMessage,
   ): Promise<NodeTypes.InstallResult | NodeTypes.RejectInstallResult> => {
     try {
+      if (data.from === this.nodeService.cfNode.publicIdentifier) {
+        logger.log(`Got our own event, not doing anything.`);
```
Comment (LayneHaber) [inline modules/node/src/appRegistry/appRegistry.service.ts]: wouldnt you want to return out here?

File: modules/node/src/swapRate/swapRate.service.ts [hunk 3]
```diff
@@ -14,49 +15,114 @@ const MEDIANIZER_ADDRESS = "0x729D19f657BD0614b4985Cf1D82531c67569197B";
...
+      oldRate = this.latestSwapRates[rateIndex].rate;
+    }
+    const tokenAddress = await this.config.getTokenAddress();
+
+    let newRate: string;
+    if (from === AddressZero && to === tokenAddress) {
+      try {
+        const bnRate = (await this.medianizer.peek())[0];
+        newRate = formatEther(bnRate);
+        logger.debug(`Got swap rate from medianizer at block ${blockNumber}: ${newRate}`);
```
Comment (LayneHaber) [inline modules/node/src/swapRate/swapRate.service.ts]: is this the only place youre using `blockNumber`? if so, we should just take it out. if not, we have to be careful of the alchemy chunking limits (i think its 10k)?

File: modules/node/src/appRegistry/appRegistry.service.ts (hunk 2 above)
Comment (rhlsthrm) [reply modules/node/src/appRegistry/appRegistry.service.ts]: Absolutely lol.

File: modules/node/src/swapRate/swapRate.service.ts (hunk 3 above)
Comment (rhlsthrm) [reply modules/node/src/swapRate/swapRate.service.ts]: But I thought the Alchemy limits are only for getting events across many blocks. What's the problem with just getting block number?

**Pull request merged**